test/test_vector.py

```diff
@@ -154,5 +215,156 @@ def test_reset(self):
         can.interfaces.vector.canlib.vxlapi.xlActivateChannel.assert_called()
 
 
+def xlGetApplConfig(
```

Comment:
Ah, you're splitting these out because all these modify the reference and have a return value? Thus in order to return those values from the function, a `side_effect` function is used? That seems to be the main rationale for this change from my understanding.

This seems reasonable to do so like what you've done, and I suppose that any tests which need different behaviour can patch it as necessary.

However, I'm just wondering whether providing this as a default implementation is any different vs what you had before, since any time we want to exercise other cases the function will probably be patched in a way specific to the test case. @felixdivo do you have any thoughts?

Reply:
Some vxlapi functions modify the arguments (mutable ctypes) that are given to the function. That's why i added these side_effects. 
I think this is better than 'dumb' mocks.

When we add new functions to the vxlapi these tests will break as no mock exists for the new function. But i don't see this as a huge problem because it encourages looking at the tests. Solving it will be pretty simple too.

Reply:
I haven't done too much with mocking, but this seems reasonable the way it is now.